### PR TITLE
Run all tests on release

### PR DESCRIPTION
**Pull request by pquentin**
@gbanasiak noticed while releasing Rally that we were only running simple unit tests. This pull request fixes the issue.

### CODE REVIEW

**Comment by pquentin** [comment]
@elastic/es-perf Is anyone here using `make test` for development? I'm using `pytest` directly, so the change to `make test` does not affect me, but maybe it affects you?

**Review by gbanasiak — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -113,7 +114,7 @@ release-checks: check-venv
 	. $(VENV_ACTIVATE_FILE); ./release-checks.sh $(release_version) $(next_version)
 
 # usage: e.g. make release release_version=0.9.2 next_version=0.9.3
-release: check-venv release-checks clean docs it
+release: check-venv release-checks clean docs test it
```
Comment (dliappis) [inline Makefile:117]: Is the intention to run the entirety of tests prior to `release`? If yes, should we include `lint` in this target too?

**Review by dliappis — COMMENTED** [review]
No objections to this change. Makes sense.
Left one question.

File: Makefile (hunk 1 above)
Comment (pquentin) [reply Makefile:117]:
```suggestion
release: check-venv release-checks clean docs lint test it
```

**Review by dliappis — APPROVED** [review]
LGTM

**Comment by pquentin** [comment]
buildkite build this please

**Pull request merged**